src/server/components/lists/types.ts

```diff
@@ -36,4 +34,41 @@ export interface ListsRequestParams {
   readNotice?: string;
   readDisclaimer?: string;
   resultsTurnaround?: string;
+  page?: string;
+}
+
+export interface PaginationResults {
```

Comment:
maybe this should be a generic? or at least extend ListRequestParams instead of copying it. My concern is that it makes it more likely that someone would change ListRequestParams, but might forget to change PaginationResults.

extension method
```
export interface PaginationResults extends ListRequestParams {..}
```


generic method, something like this: 
```
type Pagination<T> =  T & { //maybe PaginatedResult<..>?
  pagination: {
     results: {
       from: number;
       to: number;
       count: number;
       currentPage: number;
     },
     previous: paginationItem,
     next: paginationItem,
     items: PaginationItem[];
  }
}
//usage:
const result: Pagination<ListRequestParams> = {
...
}


```

Reply:
Not sure why PaginationResults would need to extend ListRequestParams here.  Typescript noob.  Can you elaborate?

Reply:
I've just had a look at the function that returns PaginationResults (`getPaginationValues`) and it doesn't actually look like any of the additional properties like serviceType, country, region etc are used at all. 

When something is defined  like this `readNotice?: string` - it means that the object can have the property `readNotice` - and if it does, it must be a string. You're not getting any ts warnings or errors on this return type because technically it's correct.. but `getPaginationValues` *never* returns any of the other keys - so let's remove them. 

As for why, let's say 5 months down the line we're changing up pagination. the type definition as it is, suggests that serviceType, country, region etc might be available so suggests there needs to be additional code to handle them, or they're added at another point. 

anyway, ramble over - ignore the extension, we don't need it, but we should remove the extraneous keys

```.ts 
{
    pagination: {
      results: {
        from,
        to,
        count,
        currentPage,
      },
      previous: {
        text: nextPrevious.previous.page.toString(),
        href: nextPrevious.previous.queryString,
      },
      next: {
        text: nextPrevious.next.page.toString(),
        href: nextPrevious.next.queryString,
      },
      items: pageItems,
    },
  };
```

Reply:
I'm such a doofus.  I didn't read it closely enough and saw the duped properties were copied from ListRequestParams.  I've now removed them.